### ISSUE TITLE
1.1.x pekko snapshots are no longer including the java 9+ classes

### ISSUE THREAD

**Issue by pjfanning** [apache/incubator-pekko#1039]
Have a look at the projects/Jdk9.scala file.

The classes that this is supposed to build appear to be no longer built.

Java 1.0.x jars still have the expected classes.

Compare
https://repository.apache.org/content/groups/snapshots/org/apache/pekko/pekko-cluster-sharding_2.13/1.0.2+26-870129ef-SNAPSHOT/

to
https://repository.apache.org/content/groups/snapshots/org/apache/pekko/pekko-cluster-sharding_2.13/1.1.0-M0+283-e597a702-SNAPSHOT/

Have a look for this Java9+ class.
org.apache.pekko.cluster.sharding.internal.jfr.JFRShardingFlightRecorder

Included in the 1.0 jar but not in the 1.1 jar.

This appears to be happening in other jars where we have Jdk9(+) classes.

**Comment by mdedetrich** [comment]
We need to bisect the commit where this happened

**Comment by He-Pin** [comment]
This is a  blocker for 1.1.0

**Comment by pjfanning** [comment]
Last pekko-cluster-sharding_2.13 snapshot to have JFR classes is 1.1.0-M0+221-959c98db-SNAPSHOT - Jan 9

Missing in next snapshot - 1.1.0-M0+224-ccce5c04-SNAPSHOT (Jan 11)

**Comment by pjfanning** [comment]
nothing obvious but https://github.com/apache/incubator-pekko/commit/b9d2cc6187e5622fee5e4d0052a4120141c5b9c4 was in Jan 9 and that sbt-osgi plugin has been causing us serious grief

**Comment by pjfanning** [comment]
@mdedetrich I reverted https://github.com/apache/incubator-pekko/commit/b9d2cc6187e5622fee5e4d0052a4120141c5b9c4 locally and it fixed the issue with missing JDK9 classes

**Comment by He-Pin** [comment]
Run with java 11
```scala
sbt:pekko-stream> show unmanagedSourceDirectories
[info] * C:\Users\hepin\IdeaProjects\incubator-pekko\stream\src\main\scala
[info] * C:\Users\hepin\IdeaProjects\incubator-pekko\stream\src\main\scala-2.13
[info] * C:\Users\hepin\IdeaProjects\incubator-pekko\stream\src\main\scala-2
[info] * C:\Users\hepin\IdeaProjects\incubator-pekko\stream\src\main\java
[info] * C:\Users\hepin\IdeaProjects\incubator-pekko\stream\src\main\scala-2.13+
sbt:pekko-stream>
```

**Comment by mdedetrich** [comment]
> @mdedetrich I reverted https://github.com/apache/incubator-pekko/commit/b9d2cc6187e5622fee5e4d0052a4120141c5b9c4 locally and it fixed the issue with missing JDK9 classes

Thanks I'll look into it

**Comment by mdedetrich** [comment]
So I already know what the likely cause is, its a fallout from https://github.com/sbt/sbt-osgi/pull/64

**Comment by He-Pin** [comment]
We will need a task to verify the jar contains these Java 9+ classes in the nightly build after packaging.

**Comment by mdedetrich** [comment]
So I can indeed confirm that its https://github.com/sbt/sbt-osgi/pull/64, if I print out what the `Compile / fullClasspath` is for `stream` (which is what contains JDK 9 classes) you can see

```
[info] * Attributed(/Users/mdedetrich/github/incubator-pekko/stream/target/scala-2.13/classes)
[info] * Attributed(/Users/mdedetrich/github/incubator-pekko/actor/target/scala-2.13/classes)
[info] * Attributed(/Users/mdedetrich/github/incubator-pekko/protobuf-v3/target/scala-2.13/stripped/stripped/pekko-protobuf-v3-assembly-1.0.2+27-46618d08-SNAPSHOT.jar)
[info] * Attributed(/Users/mdedetrich/github/incubator-pekko/protobuf-v3/target/scala-2.13/stripped/pekko-protobuf-v3-assembly-1.0.2+27-46618d08-SNAPSHOT.jar)
[info] * Attributed(/Users/mdedetrich/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.13.11/scala-library-2.13.11.jar)
[info] * Attributed(/Users/mdedetrich/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/reactivestreams/reactive-streams/1.0.4/reactive-streams-1.0.4.jar)
[info] * Attributed(/Users/mdedetrich/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/typesafe/ssl-config-core_2.13/0.6.1/ssl-config-core_2.13-0.6.1.jar)
[info] * Attributed(/Users/mdedetrich/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/typesafe/config/1.4.3/config-1.4.3.jar)
[info] * Attributed(/Users/mdedetrich/github/incubator-pekko/stream/target/scala-2.13/CompileJdk9-classes)
```

As you can see the last line is the compiled jdk9 classes which are missing

However if we print out the replacement which is `(dependencyClasspathAsJars in Compile).value.map(_.data) ++ (products in Compile)` then we get this

```
[info] * Attributed(/Users/mdedetrich/github/incubator-pekko/actor/target/scala-2.13/stripped/pekko-actor_2.13-1.0.2+27-46618d08-SNAPSHOT.jar)
[info] * Attributed(/Users/mdedetrich/github/incubator-pekko/protobuf-v3/target/scala-2.13/stripped/stripped/pekko-protobuf-v3-assembly-1.0.2+27-46618d08-SNAPSHOT.jar)
[info] * Attributed(/Users/mdedetrich/github/incubator-pekko/protobuf-v3/target/scala-2.13/stripped/pekko-protobuf-v3-assembly-1.0.2+27-46618d08-SNAPSHOT.jar)
[info] * Attributed(/Users/mdedetrich/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.13.11/scala-library-2.13.11.jar)
[info] * Attributed(/Users/mdedetrich/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/reactivestreams/reactive-streams/1.0.4/reactive-streams-1.0.4.jar)
[info] * Attributed(/Users/mdedetrich/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/typesafe/ssl-config-core_2.13/0.6.1/ssl-config-core_2.13-0.6.1.jar)
[info] * Attributed(/Users/mdedetrich/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/typesafe/config/1.4.3/config-1.4.3.jar)
```

plus

```
[info] * /Users/mdedetrich/github/incubator-pekko/stream/target/scala-2.13/classes
```

As we can see its missing the JDK 9 classes folder. I will need to do changes in sbt-osgi upstream to fix this

**Comment by mdedetrich** [comment]
So I already have a PR ready for sbt-osgi that will allow me to fix this issue, just waiting for https://github.com/sbt/sbt-osgi/pull/118 to get merged.

Should be fixed in a couple of days

**Comment by He-Pin** [comment]
after revert the osgi to 0.9.4 and publish local
![image](https://github.com/apache/incubator-pekko/assets/501740/f9fdbce1-2b8f-4e7f-83d6-dd192c83f889)
before
![image](https://github.com/apache/incubator-pekko/assets/501740/dabbcbf1-647e-4b4c-94f2-e6652795549e)

@mdedetrich yes, it's osgi plugin's problem.

**Comment by Roiocam** [comment]
It seems to be related to this #959.

**Comment by Roiocam** [comment]
> @mdedetrich I reverted [b9d2cc6](https://github.com/apache/incubator-pekko/commit/b9d2cc6187e5622fee5e4d0052a4120141c5b9c4) locally and it fixed the issue with missing JDK9 classes

@pjfanning Thanks for investigating , this problem is also led to #959.

**Comment by mdedetrich** [comment]
I am quite sure that the core problem is what I described at https://github.com/apache/incubator-pekko/issues/1039#issuecomment-1909114860 which is leading to all of the other described issues. There might be a simple solution for this that doesn't require extending sbt-osgi with additional functionality but I need to first merge + release https://github.com/sbt/sbt-osgi/pull/121 .

**Comment by Roiocam** [comment]
@pjfanning I think this has been solve by #1047

**Comment by pjfanning** [comment]
> @pjfanning I think this has been solve by #1047

I disagree - #1040 is to test #1039 and #1039 is not even fixed yet. There is zero proof that this is a fix for #1040.

**Comment by Roiocam** [comment]
> I disagree - #1040 is to test #1039 and #1039 is not even fixed yet. There is zero proof that this is a fix for #1040.

My last repeated comment was published in the wrong ISSUE. This comment was published under #1039, that is, we have fixed the compile of the JDK9 class on package task.
